sdk/wallet-adapter/wallet-kit-core/src/index.ts

```diff
@@ -47,17 +44,22 @@ export interface WalletKitCoreState extends InternalWalletKitCoreState {
   isError: boolean;
 }
 
+type WithOptionalParams<
```

Comment:
Rather than make a specific version of this, I'd actually just make this generic:

```typescript
type OptionalProperties<
  T extends Record<string, unknown>,
  U extends keyof T,
> = Omit<T, U> & Partial<Pick<T, U>>;
```

Then at the callsite just doing:
```typescript
OptionalProperties<SuiSignTransactionInput, 'account' | 'chain'>
```